graviton/invariant.py

```diff
@@ -10,14 +11,47 @@
 )
 
 
+class PredicateKind(Enum):
+    Constant = "constant"
+    CaseMap = "case map"
+    ExactMatch = "exact match"
+    RangeMatch = "range match"
+    IdenticalPair = "identical"
+
+
 INVARIANT_TYPE = Union[
     PyTypes,
     Dict[Sequence[PyTypes], PyTypes],
     Callable[[PyTypes], PyTypes],
     Callable[[PyTypes], bool],
+    PredicateKind,  # currently represents the equality of two dry-run executable functions
```

Comment:
I assume that, all the previous elements in union, namely, 
```python
    PyTypes,
    Dict[Sequence[PyTypes], PyTypes],
    Callable[[PyTypes], PyTypes],
    Callable[[PyTypes], bool],
```
in `get_kind` get mapped to the first four in `PredicateKind`, i.e.,
```python3
    Constant = "constant"
    CaseMap = "case map"
    ExactMatch = "exact match"
    RangeMatch = "range match"
```
with the `IdenticalPair` introduction in this PR. Does that sound correct to you?

Reply:
yes, except that exact match's type was poorly specified